ui/src/components/Navbar.js

```diff
@@ -0,0 +1,102 @@
+//@flow
+import React, { useEffect, useLayoutEffect, useRef, useState } from 'react';
+import { useTypedSelector } from '../hooks';
+import { updateAPIConfigAction } from '../ducks/config';
+import { useDispatch } from 'react-redux';
+
+function useWebComponent(src: string) {
+  useEffect(() => {
+    const body = document.body;
+    // $flow-disable-line
+    const element = [...(body?.querySelectorAll('script') || [])].find(
+      // $flow-disable-line
+      (scriptElement) => scriptElement.attributes.src?.value === src,
+    );
+
+    if (!element && body) {
+      const scriptElement = document.createElement('script');
```

Comment:
When we will switch to module federation if in some cases we want to use this hook to import another web component we then cannot take any assumption about the target ui implementation that could have manually imported the same web component bundle without the expected id. In this case it would import the script a second time redefining the web component and leading to some errors. 
IMO it seems safer to check if the bundle is already imported to avoid constraining web component bundles imports.

Reply:
OK, I guess you could give this `id` as an argument to the hook, but this approach works fine so let's keep it.